File: scripts/check_claim_work_hours.py
import sys
sys.path.insert(0, r'C:\Project\SambioHRR\excel-upload-server')
from core.db_manager import DatabaseManager

db_path = r'C:\Project\SambioHRR\sambio_human.db'
db = DatabaseManager(db_path)
conn = db.get_connection()
cursor = conn.cursor()

print('=== 1-6월 claim_data 근무시간 패턴 확인 ===\n')

# 1. 공휴일/주말 패턴 확인
print('1. 날짜별 평균 근무시간 (1-6월)')
cursor.execute("""
    SELECT 
        날짜,
        COUNT(*) as 직원수,
        AVG(근무시간) as 평균근무시간,
        SUM(CASE WHEN 근무시간 = 0 THEN 1 ELSE 0 END) as 근무시간0명,
        SUM(CASE WHEN 근무시간 = 8 THEN 1 ELSE 0 END) as 근무시간8명,
        SUM(CASE WHEN 휴가_연차 > 0 THEN 1 ELSE 0 END) as 휴가자수
    FROM claim_data
    WHERE 날짜 >= '2025-01-01' AND 날짜 <= '2025-01-31'
    GROUP BY 날짜
    ORDER BY 날짜
""")

for row in cursor.fetchall():
    날짜, 직원수, 평균근무시간, 근무시간0명, 근무시간8명, 휴가자수 = row
    print(f'{날짜}: {직원수}명, 평균 {평균근무시간:.1f}h, 0h={근무시간0명}명, 8h={근무시간8명}명, 휴가={휴가자수}명')

print('\n2. 특정 직원의 1월 근무시간 상세')
cursor.execute("""
    SELECT 날짜, 근무시간, 휴가_연차, 출장, leave_type
    FROM claim_data
    WHERE 사번 = (SELECT 사번 FROM claim_data WHERE 날짜 >= '2025-01-01' LIMIT 1)
    AND 날짜 >= '2025-01-01' AND 날짜 <= '2025-01-31'
    ORDER BY 날짜
""")

for row in cursor.fetchall():
    print(f'{row[0]}: 근무={row[1]}h, 휴가={row[2]}h, 출장={row[3]}h, 휴가타입={row[4]}')

print('\n3. 근무시간 분포 (1-6월 전체)')
cursor.execute("""
    SELECT 
        근무시간,
        COUNT(*) as 건수,
        COUNT(DISTINCT 사번) as 직원수,
        ROUND(COUNT(*) * 100.0 / (SELECT COUNT(*) FROM claim_data WHERE 날짜 >= '2025-01-01' AND 날짜 <= '2025-06-30'), 2) as 비율
    FROM claim_data
    WHERE 날짜 >= '2025-01-01' AND 날짜 <= '2025-06-30'
    GROUP BY 근무시간
    ORDER BY 근무시간
""")

for row in cursor.fetchall():
    print(f'근무시간 {row[0]}h: {row[1]:,}건 ({row[2]}명, {row[3]}%)')

db.close()

File: scripts/check-claim-work-hours.ts
#!/usr/bin/env tsx
import Database from 'better-sqlite3'
import path from 'path'

const db = new Database(path.join(process.cwd(), 'sambio_human.db'), { readonly: true })

console.log('=== 1-6월 claim_data 근무시간 패턴 확인 ===\n')

// 1. 1월 날짜별 통계
console.log('1. 1월 날짜별 근무시간 통계:')
const janStats = db.prepare(`
  SELECT
    근무일자,
    COUNT(*) as 직원수,
    AVG(CAST(근무시간 AS REAL)) as 평균근무시간,
    SUM(CASE WHEN CAST(근무시간 AS REAL) = 0 THEN 1 ELSE 0 END) as 근무시간0명,
    SUM(CASE WHEN CAST(근무시간 AS REAL) = 8 THEN 1 ELSE 0 END) as 근무시간8명
  FROM claim_data
  WHERE 근무일자 >= '2025-01-01' AND 근무일자 <= '2025-01-31'
  GROUP BY 근무일자
  ORDER BY 근무일자
`).all() as any[]

janStats.forEach((row: any) => {
  const avg = Number(row.평균근무시간 || 0).toFixed(1)
  console.log(`${row.근무일자}: ${row.직원수}명, 평균 ${avg}h, 0h=${row.근무시간0명}명, 8h=${row.근무시간8명}명`)
})

// 2. 근무시간 분포
console.log('\n2. 근무시간 값 분포 (1-6월):')
const distribution = db.prepare(`
  SELECT
    근무시간,
    COUNT(*) as 건수,
    ROUND(COUNT(*) * 100.0 / (SELECT COUNT(*) FROM claim_data WHERE 근무일자 >= '2025-01-01' AND 근무일자 <= '2025-06-30'), 2) as 비율
  FROM claim_data
  WHERE 근무일자 >= '2025-01-01' AND 근무일자 <= '2025-06-30'
  GROUP BY 근무시간
  ORDER BY CAST(근무시간 AS REAL)
`).all() as any[]

distribution.forEach((row: any) => {
  console.log(`${row.근무시간}: ${row.건수.toLocaleString()}건 (${row.비율}%)`)
})

db.close()
